packagesFromOverlaysBuilderConstructor.nix

```diff
@@ -0,0 +1,111 @@
+{ flake-utils-plus }: let
+
+ overlayFromPackagesBuilderConstructor = channels: let
+   # channels: channels.<name>.overlays
+
+   overlayFromPackagesBuilder = pkgs:
+    /**
+    Synopsis: overlayFromPackagesBuilder _pkgs_
+
+    pkgs:     pkgs.<system>.<tree>
+
+    Returns valid packges that have been defined within an overlay so they
+    can be shared via _self.packages_ with the world. This is especially useful
+    over sharing one's art via _self.overlays_ in case you have a binary cache
+    running from which third parties could benefit.
+
+    First, flattens an arbitrarily nested _pkgs_ tree for each system into a flat
+    key in which nesting is aproximated by a "/" (e.g. "development/kakoune").
+    Also filter the resulting packages for attributes that _trivially_ would
+    fail flake checks (broken, system not supported or not a derivation).
+
+    Second, collects all overlays' packages' keys of all channels into a flat list.
+
+    Finally, only passes packages through the seive that are prefixed with a top level
+    key exposed by any of the overlays. Since overlays override (and do not emrge)
+    top level attributes, by filtering on the prefix, an overlay's entire packages
+    tree will be correctly captured.
+
+    Example:
+
+    pkgs'.<system> = {
+      "development/kakoune" = { ... };
+      "development/vim" = { ... };
+    };
+
+    overlays' = [
+      "development"
+    ];
```

Comment:
It would be cool to make this typesafe, something along the lines of:
```
    overlays' = [
      xyz.development
    ];
```
Just some food for thought, dont mind me :)

Reply:
It is in fact stringly typed on purpose for the (string) prefix matching to work after pkgs' tree has been flattened. I also wouldnt be able to tell what `xyz` could be, since overlays  define top level attributes on the `pkgs` namespace.